src/main/java/com/mathworks/ci/RunMatlabTestsStep.java

```diff
@@ -0,0 +1,185 @@
+package com.mathworks.ci;
+/**
+ * Copyright 2020 The MathWorks, Inc.
+ *  
+ */
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Copyright 2020 The MathWorks, Inc.
+ *  
+ */
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.io.FilenameUtils;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import com.google.common.collect.ImmutableSet;
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+public class RunMatlabTestsStep extends Step {
+    
+    private String testResultsPDF;
+    private String testResultsTAP;
+    private String testResultsJUnit;
+    private String codeCoverageCobertura;
+    private String testResultsSimulinkTest;
+    private String modelCoverageCobertura;
+    private static final String PDF_REPORT_PATH = "PDFReportPath";
```

Comment:
Do these Strings have to be at class scope? I saw them being used in just 1 method.

Reply:
Makes sense. i have removed it from the class scope. and used them directly in the method.